### PR TITLE
Set GOMEMLIMIT on Istiod

### LINKED ISSUE #40676: Set GOMEMLIMIT
New feature in 1.19, should be useful for us

https://github.com/KimMachineGun/automemlimit _might_ allow auto-detect in k8s, but needs  testing. We can also do it via helm/downward api/etc, but may be more complex

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/40676

**Please provide a description of this PR:**

### CODE REVIEW

**Comment by ericvn** [comment]
I suspect an extra file was added in the commit.

File: releasenotes/notes/gomemlimit.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+issue:
+- 40676
+
+releaseNotes:
+- |
+  **Added** an automatically set GOMEMLIMIT to `istiod` to reduce the risk of out-of-memory issues.
```
Comment (ericvn) [inline releasenotes/notes/gomemlimit.yaml:9]: nit: I was a little confused the first time I read this thinking the initial `an` should maybe be `and`. But after reading it a few times, `an` does make sense.  Could we maybe remove the `an automatically set` and not lose anything? or maybe Added GOMEMLIMIT, set to limits.memory, to `istiod` to reduce...

**Review by ericvn — APPROVED** [review]
File removed. Thanks.

**Review by zirain — APPROVED** [review]

**Review by hanxiaop — APPROVED** [review]

File: releasenotes/notes/gomemlimit.yaml (hunk 1 above)
Comment (rabbbit) [inline releasenotes/notes/gomemlimit.yaml:9]: IMO, GOMEMLIMIT is not really a tool to reduce the out-of-memory issues. If you're out of memory, you are less likely to OOM, but instead, you'll see performance degradation - as you get closer to the limit, GC will try harder not to breach it. So you're trading one issue for another.

GOMEMLIMIT is, however, a great CPU utilization reduction tool for services that are very far from the "physical" memory limit though - you'll run GC less often.

**Pull request merged**